src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Executor} from '../Backend/Executor';
+
+interface ExecutionEnv {
+  name(): string;
+  envInfo(): string;
+  getExecutors(): Executor[];
+  getExecutor(name: string): Executor|undefined;
+  setExecutor(name: string, executor: Executor): boolean;
+}
+
+// This is a base ExeuctionEnv.
+// on New ExecutionEnv implements, need to add 3things
+// 1. ExecutionEnv itself
+// 2. ExecutionEnv Creation function
+//    This will return class for Env that extends ExecutionEnvBase
+// 3. ExecutionEnv list get function
+//    This will return Commend that execute on host and return string array of device list,
+//    could split with '\n'.
```

Comment:
Then, why `getConnectableEnvs` is not a class method, but defined in each module?

Reply:
Plus, why does it returns `Command`, not returning the device array after executing the command by itself?

Reply:
> Then, why `getConnectableEnvs` is not a class method, but defined in each module?

if we define this in class, we need to create Env before it really exists. so, I thought module could check this.

Reply:
> Plus, why does it returns `Command`, not returning the device array after executing the command by itself?

I thought it could run on remote host, then we should run those in `ExecutionEnvManager` object.